model/src/main/kotlin/spdx/SpdxExpression.kt

```diff
@@ -58,13 +62,23 @@ sealed class SpdxExpression {
             return visitor.visit(parser.licenseExpression())
         }
     }
+
+    /**
+     * Return all valid SPDX licenses contained in this expression. LicenseRefs and invalid licenses are ignored.
+     *
+     * TODO: The result should also contain SPDX license exceptions associated to the licenses.
+     */
+    abstract fun spdxLicenses(): EnumSet<SpdxLicense>
 }
 
 data class SpdxCompoundExpression(
         val left: SpdxExpression,
         val operator: SpdxOperator,
         val right: SpdxExpression
 ) : SpdxExpression() {
+    override fun spdxLicenses(): EnumSet<SpdxLicense> =
+            EnumSet.copyOf(left.spdxLicenses()).apply { addAll(right.spdxLicenses()) }
```

Comment:
How about this instead (you don't need the explicit return type then anymore):

    override fun spdxLicenses() = enumSetOf(*left.spdxLicenses().toTypedArray(), *right.spdxLicenses().toTypedArray())

Reply:
I don't like creating intermediate objects just for the sake of syntax sugar, but I made another extension function `EnumSet.plus` in the meantime which I now added to this PR.